### PR TITLE
Implement new front end design

### LINKED ISSUE #1: Get a real design

### PR DESCRIPTION
**Pull request by TammyTee**
## Summary

This resolves #1 🎉 

**Notable changes (some outside of FED)**

* Ran Headwind on the project to organize the tailwind classes
* Added a `skill_level` column to modules so that they can be organized on the listing page by beginner, intermediate, and advanced
* Changed the module listing page to list all module **NOT** in a users track when they're logged in
* Add filter to Modules in Nova to filter by skill level
* Add new layout that has no nav to use for auth views
* Organize Vue component imports for those that are used globally 

### Question(s)
* Is `learn.blade.php` still in use?

### CODE REVIEW

**Comment by mattstauffer** [comment]
> Is learn.blade.php still in use?

Hm I'd lean toward no? But let's see lol

File: resources/views/partials/resource-on-module-page-with-border.blade.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<li class="inline-flex w-full pt-8 pl-6 pr-5 mt-6 border-t-2 first:mt-0 first:pt-0 first:border-0 lg:px-12">
+    @auth
+        @if (Auth::user()->hasTrack() && Auth::user()->track->modules->contains($module->id))
+            <completed-checkbox
+                class="mr-5"
+                :initial-is-completed="{{ $completedResources->contains($resource->id) ? 'true' : 'false' }}"
+                type="{{ $resource->getMorphClass() }}"
+                id="{{ $resource->id }}"
+                ></completed-checkbox>
```
Comment (damiani) [inline resources/views/partials/resource-on-module-page-with-border.blade.php:9]:
```suggestion
            />
```

File: resources/views/partials/resource-on-module-page.blade.php [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
-        type="{{ $resource->getMorphClass() }}"
-        id="{{ $resource->id }}"
-        ></completed-checkbox>
+        @if (Auth::user()->hasTrack() && Auth::user()->track->modules->contains($module->id))
+        <completed-checkbox
+            class="mr-5"
+            :initial-is-completed="{{ $completedResources->contains($resource->id) ? 'true' : 'false' }}"
+            type="{{ $resource->getMorphClass() }}"
+            id="{{ $resource->id }}"
+            ></completed-checkbox>
```
Comment (damiani) [inline resources/views/partials/resource-on-module-page.blade.php:9]:
```suggestion
        />
```

File: resources/views/vendor/nova/partials/footer.blade.php [hunk 3]
```diff
@@ -1,5 +1,5 @@
-<p class="mt-8 text-center text-xs text-80">
-    <a href="https://nova.laravel.com" class="text-primary dim no-underline">Laravel Nova</a>
+<p class="mt-8 text-xs text-center text-80">
+    <a href="https://nova.laravel.com" class="no-underline text-primary dim">Laravel Nova</a>
```
Comment (damiani) [inline resources/views/vendor/nova/partials/footer.blade.php:2]: We don't need to make changes to Nova's vendor files

File: resources/views/vendor/nova/partials/user.blade.php [hunk 4]
```diff
@@ -14,7 +14,7 @@ class="rounded-full w-8 h-8 mr-3"
 <dropdown-menu slot="menu" width="200" direction="rtl">
     <ul class="list-reset">
         <li>
-            <a href="{{ route('nova.logout') }}" class="block no-underline text-90 hover:bg-30 p-3">
+            <a href="{{ route('nova.logout') }}" class="block p-3 no-underline text-90 hover:bg-30">
```
Comment (damiani) [inline resources/views/vendor/nova/partials/user.blade.php:17]: We don't need to make changes to Nova's vendor files

File: app/Module.php [hunk 5]
```diff
@@ -13,6 +13,16 @@ class Module extends Model implements Completable
...
+    const BEGINNER_SKILL_LEVEL = 'beginner';
+    const INTERMEDIATE_SKILL_LEVEL = 'intermediate';
+    const ADVANCED_SKILL_LEVEL = 'advanced';
+
+    const SKILL_LEVELS = [
+        self::BEGINNER_SKILL_LEVEL,
+        self::INTERMEDIATE_SKILL_LEVEL,
+        self::ADVANCED_SKILL_LEVEL,
+    ];
+
```
[the remark this answers is not in the document]
Comment (TammyTee) [reply app/Module.php:25]: I believe I know what you mean here but would like to quick chat about it

**Comment by TammyTee** [comment]
Hey @damiani ! I corrected all of the edits excluding the two I haven't marked as resolved. Let me know when you have a quick sec to chat about the things you noted so I can knock these last two things out.

Thanks for reviewing this!! 😄

**Comment by TammyTee** [comment]
@damiani The module listing changes have been completed

**What changed:** 
* created a `ModuleListing` component for ease of switching between designs when a user is a guest or logged in
* added a toggle for logged in users to filter between the modules they're seeing
* added a badge to cards to indicate if a module is completed

**Screenshot**
image

File: resources/js/components/Tabs/Tabs.vue [hunk 6]
```diff
@@ -75,12 +84,12 @@ export default {
                 return;
             }
 
-            if($(window).width() >= 992) {
+            if(window.innerWidth >= 992) {
```
Comment (damiani) [inline resources/js/components/Tabs/Tabs.vue:87]:
```suggestion
            if (window.innerWidth >= 992) {
```

**Pull request merged**